addons/account/models/account_order.py

```diff
@@ -0,0 +1,229 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import Command, _, api, fields, models
+from odoo.tools import float_is_zero
+
+
+class AccountOrder(models.AbstractModel):
+    # This class enables downpayment support for models that can have down payments on them
+    _name = 'account.order'
+    _description = 'Account Order'
+
+    amount_to_invoice = fields.Monetary("Amount to invoice", store=True, compute='_compute_amount_to_invoice')
+    amount_invoiced = fields.Monetary("Already invoiced", compute='_compute_amount_invoiced')
+
+    # To override
+    company_id = fields.Many2one('res.company')
+    currency_id = fields.Many2one('res.currency')
+    fiscal_position_id = fields.Many2one('account.fiscal.position', string='Fiscal Position')
+    partner_id = fields.Many2one('res.partner')
+
+    amount_total = fields.Monetary()
+
+    invoice_ids = fields.Many2many('account.move')
```

Comment:
I thought oco suggested to use `account_move` or `move` instead of invoice ? (this commit is actually valid for all other usages of invoices)